tokio/src/net/tcp/listener.rs

```diff
@@ -234,12 +281,26 @@ impl TcpListener {
     /// from a future driven by a tokio runtime, otherwise runtime can be set
     /// explicitly with [`Runtime::enter`](crate::runtime::Runtime::enter) function.
     #[track_caller]
-    pub fn from_std(listener: net::TcpListener) -> io::Result<TcpListener> {
+    pub fn from_tcp_unchecked(listener: net::TcpListener) -> io::Result<TcpListener> {
```

Comment:
The documentation of this method should have a bigger focus on the fact that this is an unchecked conversion. Please see [`from_file_unchecked`][1] for inspiration.

[1]: https://docs.rs/tokio/latest/tokio/net/unix/pipe/struct.Sender.html#method.from_file_unchecked

Reply:
Is the the fact that the underlying file descriptor might not be set to `non-blocking` the _only_ behavioral change? Even if so, this seems like reasonable naming.

(I'm assuming that's what you're referring to.)

Reply:
In my original suggestion it would also return an error if the underlying fd is not actually a tcp socket, but we can debate whether we want that.